src/charm.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical
+# See LICENSE file for licensing details.
+
+import logging
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresqlOperatorCharm(CharmBase):
+    """Charmed Operator for the PostgreSQL database."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    def _on_config_changed(self, _):
+        """Handle the config-changed event"""
+        # Get the postgresql container so we can configure/manipulate it
+        container = self.unit.get_container("postgresql")
+        # Create a new config layer
+        layer = self._postgresql_layer()
+
+        if container.can_connect():
+            # Get the current config
+            services = container.get_plan().to_dict().get("services", {})
```

Comment:
I believe you can access the `services` attribute directly with `container.get_plan().services`. See [here](https://github.com/canonical/operator/blob/aed34f9bcebf24effb6f74fab80c6c7fdfc87dda/ops/pebble.py#L433).

Reply:
Thanks for the suggestion, Daniela! It makes the code better. I applied it.